src/Simulation/qdk_sim_rs/test-qdk-sim-rs.ps1

```diff
@@ -0,0 +1,60 @@
+& (Join-Path $PSScriptRoot ".." ".." ".." "build" "set-env.ps1");
+
+$script:allOk = $true;
+
+Push-Location $PSScriptRoot
+    # Start with the quick check first and make sure that Rust sources
+    # meet formatting and style guide rules.
+    cargo fmt -- --check
+    $script:allOk = $script:allOk -and $LASTEXITCODE -eq 0;
+
+    # Check linting rules defined by clippy, a linting tool provided with the
+    # Rust toolchain. Please see https://github.com/rust-lang/rust-clippy
+    # and https://rust-lang.github.io/rust-clippy/master/index.html
+    # for more information.
+    # If there's a false positive, that check should be explicitly disabled
+    # at the point where the false positive occurs with an explanation as to
+    # why it's OK.
+    cargo clippy -- -D warnings
+    $script:allOk = $script:allOk -and $LASTEXITCODE -eq 0;
+
+    # If running in CI, use cargo2junit to expose unit tests to the
+    # PublishTestResults task.
+    if ("$Env:TF_BUILD" -ne "") {
+        cargo install cargo2junit
+        $testJson = cargo +nightly test -- -Z unstable-options --format json;
+        $script:allOk = $script:allOk -and $LASTEXITCODE -eq 0;
+
+        $testJson `
+            | cargo2junit `
+            <# We use this name to match the *_results.xml pattern that is used
+               to find test results in steps-wrap-up.yml. #> `
+            | Out-File -FilePath opensim_results.xml -Encoding utf8NoBOM
+    } else {
+        # Outside of CI, show human-readable output.
+        cargo +nightly test
```

Comment:
I've gated the `cargo clean` call so that it doesn't run in local builds, as per your earlier feedback. That said, I agree that in CI builds this will cause the crate to get built twice. I don't know how to avoid that in our current e2e builds given the hard disk space limitations and the order in which we call build and test commands. Would love to get @anpaz's input as to what may be a good way of cutting down on extraneous builds may be.